storage/remote/storage_test.go

```diff
@@ -133,3 +123,34 @@ func TestIgnoreExternalLabels(t *testing.T) {
 	err := s.Close()
 	require.NoError(t, err)
 }
+
+// baseRemoteWriteConfig copy values from global Default Write config
+// to avoid change global state and cross impact test execution
+func baseRemoteWriteConfig(host string) *config.RemoteWriteConfig {
+	return &config.RemoteWriteConfig{
+		RemoteTimeout:    config.DefaultRemoteWriteConfig.RemoteTimeout,
+		QueueConfig:      config.DefaultRemoteWriteConfig.QueueConfig,
+		MetadataConfig:   config.DefaultRemoteWriteConfig.MetadataConfig,
+		HTTPClientConfig: config.DefaultRemoteWriteConfig.HTTPClientConfig,
```

Comment:
Do you need to copy all of these or would code like the following work?
```
cfg := config.DefaultRemoteWriteConfig
cfg.URL = &common_config.URL{
	URL: &url.URL{
		Host: host,
	},
}
return &cfg
```
I think that would still create a copy of `config.DefaultRemoteWriteConfig` before editing the URL and would be safe. That way any new default fields would be picked up automatically in the future. This could also be done for the remote read config.

Reply:
Good catch! that's is better indeed. I've just updated and tested that